core/domain/email_manager.py

```diff
@@ -376,18 +376,20 @@ def send_mail_to_admin(email_subject, email_body):
         body, body.replace('\n', '<br/>'), bcc_admin=False)
 
 
-def send_post_signup_email(user_id):
+def send_post_signup_email(user_id, test_for_duplicate_email=False):
     """Sends a post-signup email to the given user.
 
     Raises an exception if emails are not allowed to be sent to users (i.e.
     feconf.CAN_SEND_EMAILS is False).
 
     Args:
         user_id: str. User ID of the user that signed up.
+        test_for_duplicate_email: bool. For testing duplicate emails.
     """
 
     for key, content in SIGNUP_EMAIL_CONTENT.value.items():
-        if content == SIGNUP_EMAIL_CONTENT.default_value[key]:
+        if content == SIGNUP_EMAIL_CONTENT.default_value[key] and (
+                not test_for_duplicate_email):
```

Comment:
Unfortunately this is not the right way we want to do it either.

Reply:
You want to target the test file so that it doesn't call `_send_email`, but rather this function.